File: ObjC-SampleApp/SideEngineSample/SideEngineSample/TestViewModeController.m
//
//  TestViewModeController.m
//  SideEngineSample
//
//  Created by PhoenixiOS-1 on 17/11/22.
//

#import "TestViewModeController.h"

@interface TestViewModeController ()

@end

@implementation TestViewModeController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self.navigationController setNavigationBarHidden:YES animated:NO];
}
-(void)backToHome {
    [[BBSideEngineManager shared] resumeSideEngine];
    [self.navigationController popViewControllerAnimated:NO];
}
- (IBAction)closeTapped:(UIButton *)sender {
    [self.navigationController popViewControllerAnimated:YES];
}
/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Swift-SampleApp/SideEngineSample/SideEngineSample/TestViewModeController.swift
//
//  TestViewModeController.swift
//  SideEngineSample
//
//  Created by Phoenix on 4/4/22.
//

import UIKit
import BBSideEngine
class TestViewModeController: UIViewController {
 
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    func backToHome() {
        BBSideEngineManager.shared.resumeSideEngine() //You need to resume side engine when go to back screen
        self.navigationController?.popViewController(animated: false)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        navigationController?.setNavigationBarHidden(true, animated: false)
        
    }
    
    @IBAction func closeTapped() {
        self.navigationController?.popToRootViewController(animated: true)
    }
}
